### PR TITLE
GEODE-1793 LocatorDUnitTest.testStartTwoLocatorsOneWithSSLAndTheOther…

### PR DESCRIPTION
**Pull request by bschuchardt**
This was a product issue.  When the locator using plain-text sockets is
contacted by a TcpClient using SSL the locator often just closes the socket.
On some platforms this causes a SSLHandshakeException but on others it
just causes a "SocketException: connection reset".  Writing some text to
the socket forces the TcpClient to get a SSLException (which is the superclass
of SSLHandshakeException).

The test class is still marked as Flaky due to GEODE-2542.

I deleted one of the tests in LocatorDUnitTest as it wasn't doing any useful validation and really served no purpose.

I also increased the joinTimeout in this test.  The original 1-second timeout was intended to make the tests run faster but I think it's probably the source of some of the flaky-ness in this set of tests.  Some of them were also overriding the joinTimeout established by the DUnit framework, so that was actually a bad thing to be doing.  The tests all run in a few seconds with the default joinTimeout setting anyway.

### CODE REVIEW

**Comment by bschuchardt** [comment]
There are "spotless" problems I'm cleaning up, and I'm removing the commented out code from the test.

File: geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java [hunk 1]
```diff
@@ -360,6 +360,13 @@ private void processRequest(final Socket sock) {
           versionOrdinal = (short) GOSSIP_TO_GEMFIRE_VERSION_MAP.get(gossipVersion);
         } else {
           // Close the socket. We can not accept requests from a newer version
+          try {
+            sock.getOutputStream().write("unknown protocol version".getBytes());
```
Comment (galen-pivotal) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java]: Is there any risk of this being interpreted by anything other than garbage on the other side?

File: geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java [hunk 2]
```diff
@@ -77,9 +77,9 @@
    * <p>
    * This should be incremented if the gossip message structures change
    * <p>
-   * 1000 - gemfire 5.5 - using java serialization 1001 - 5.7 - using DataSerializable and
```
Comment (galen-pivotal) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java]: Did spotless break this? 😮

Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java]: I didn't check but that's my assumption.  It likes the comment with the HTML breaks in place.

File: geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java (hunk 1 above)
Comment (bschuchardt) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpServer.java]: This situation should never happen.  When we fetch the protocol version of a Locator we use v5.7.  Then we use the locator's protocol version in future communications.  We will never send a higher version number than the locator knows how to handle.

For SSL, this text is nothing like the response to a HelloClient packet.  On machines I've tested on it results in either a SSLHandshakeException or a SSLException hinting that the other end might be using plain text sockets.

**Comment by bschuchardt** [comment]
remote: geode git commit: GEODE_1793 spotless fixes and removal of dead code
remote: geode git commit: GEODE-1793 LocatorDUnitTest.testStartTwoLocatorsOneWithSSLAndTheOtherNonSSL
To https://git-wip-us.apache.org/repos/asf/geode.git
   c09a856..4112204  develop -> develop

**Pull request closed without merging**